finesse/hardware/plugins/stepper_motor/st10_controller.py

```diff
@@ -182,10 +172,20 @@ def __init__(self, port: str, baudrate: int = 9600, timeout: float = 5.0) -> Non
         SerialDevice.__init__(self, port, baudrate)
 
         self._reader = _SerialReader(self.serial, timeout)
-        self._reader.async_read_completed.connect(self._send_move_end_message)
+        self._reader.async_read_completed.connect(self._on_initial_move_end)
         self._reader.read_error.connect(self.send_error_message)
         self._reader.start()
 
+        self._init_error_timer = QTimer()
+        """A timer to raise an error if the motor takes too long to move."""
+        self._init_error_timer.setInterval(10000)  # 10 seconds
+        self._init_error_timer.setSingleShot(True)
+        self._init_error_timer.timeout.connect(
+            lambda: self.send_error_message(
+                RuntimeError("Timed out waiting for motor to move")
```

Comment:
Ok, so this is raised and then... what? How it is the current (stuck) connection handled?

Reply:
It is handled the same way all device errors are atm: the user gets an error message in a dialog and the device is closed. I've never seen this happen, but I think it makes sense to check it rather than wait indefinitely.

Reply:
Note that this is just for the initial move, when we're homing the motor.

You can actually ask the motor controller if a drive error has occurred, which would be a useful thing to do the rest of the time when we're trying to move the motor, but that's a separate job.